### DISCUSSION TITLE
Extracted data consistently show bimodal distributions and gaps around zero association?

### DISCUSSION THREAD

**Discussion by tarjeiw** [MRCIEU/TwoSampleMR#440] [Q&A] [answered]
I should preface by saying that I am new to TwoSampleMR. However, I have used quite some time trying to figure out this issue before posting here. I am using TwoSampleMR to extract data on years of schooling (UK biobank ID: eu-b-1239, ieu-b-1001) and income (ukb-b-7408) using the code below. Regardless of exposure or outcome I extract (I have additionally tried other exposures and outcomes), I keep getting data that at least to me seem to contain concerning characteristics. Especially the two following characteristics make me think that I am doing something wrong and would be happy for input:

1. Distributions of instrument-exposure and instrument-outcome associations persistently have a bimodal distribution with a drop around 0.
2. Scatterplots of instrument-exposure and instrument-outcome assocations (not surprisingly given 1) show a linear relationship but with a gap around 0; thus, there are two clouds of data points separated by a gap.

Have anyone else experienced these issues and how did you solve this? 

Here is the code I have used which seem consistent with other examples I have seen.
```
### Extract the instruments from the schooling GWAS (ID: 'ieu-b-1239')
exposure_dat <- extract_instruments(c('ieu-a-1239'))
exposure_dat <- clump_data(exposure_dat)
### Extract outcome: household income (ID: 'ukb-b-7408')
outcome_dat <- extract_outcome_data(exposure_dat$SNP, c('ukb-b-7408'),
                  proxies = 1, rsq = 0.8, align_alleles = 1,
                  palindromes = 1, maf_threshold = 0.3)
## Harmonize the exposure and outcome data
H_data <- harmonise_data(exposure_dat, outcome_dat, action = 2)
```

**Comment by mightyphil2000** [comment, marked as answer]
By default, extract_instruments(c('ieu-a-1239')) will extract associations with P values < 5e^-8, so you shouldn't expect to see any exposure effect sizes of zero.

**Reply by tarjeiw** [reply]
Thank you, that makes perfect sense!
